e2e/validation.go

```diff
@@ -114,8 +115,8 @@ func ValidateCommonLinux(ctx context.Context, s *Scenario) {
 		ValidateKubeletNodeIP(ctx, s)
 	}
 
-	// TODO: Add support for AKSNodeConfig
-	if s.Runtime.NBC != nil && s.Runtime.NBC.AgentPoolProfile.LocalDNSProfile != nil {
+	// localdns is not supported on 1804, VHDAzureLinux3OSGuard, privatekube and VHDUbuntu2204Gen2ContainerdAirgappedK8sNotCached.
+	if s.VHD != config.VHDUbuntu1804Gen2Containerd && s.VHD != config.VHDUbuntu2204Gen2ContainerdPrivateKubePkg && s.VHD != config.VHDUbuntu2204Gen2ContainerdAirgappedK8sNotCached && s.VHD != config.VHDAzureLinux3OSGuard {
```

Comment:
This doesn't include arm64.

I think you can add another property on `Image` struct instead. Like `UnsupportedLocalDNS`